docs/modules/serialization/pages/compact-serialization.adoc

```diff
@@ -35,368 +35,1501 @@ a class to implement a particular interface. Serializers might be implemented an
+serializer must be written to support such field types. In that serializer, the inner array
+type must be defined as a separate class which stores the array type as a field. Then, the
+array of array type can be serialized/deserialized as an array of that separate class.
+
+For Java APIs, a zero-config Compact serializer uses reflection to read and write to fields of
+objects, regardless of whether those fields are public. If you use Java 9 onwards, you must
+enable reflective access for packages of your module by using the `opens` statement in the
+module declaration:
+
+[source,text]
+----
+module org.example.Foo {
+    opens org.example to com.hazelcast.core;
+}
+----
+
+== Schema Evolution
+
+Compact serialization permits schemas and classes to evolve by adding or removing fields, or
+by changing the types of fields. More than one version of a class may live in the same cluster
+and different clients or members might use different versions of the class.
+
+Hazelcast handles the versioning internally. So, you don't have to change anything in the classes
+or serializers apart from the added, removed, or changed fields.
+
+Hazelcast achieves this by identifying each version of the class by a unique fingerprint. Any change
+in a class results in a different fingerprint. Hazelcast uses a 64-bit
+https://en.wikipedia.org/wiki/Rabin_fingerprint[Rabin Fingerprint] to assign identifiers to schemas,
+which has an extremely low collision rate.
+
+Different versions of the schema with different identifiers are replicated in the cluster and can be
+fetched by clients or members internally. That allows old readers to read fields of the classes they
+know when they try to read data serialized by a new writer. Similarly, new readers might read
+fields of the classes available in the data, when they try to read data serialized by an old writer.
+
+Assume that the two versions of the following `Employee` class lives in the cluster.
+
+[tabs]
+====
+
+Java::
++
+--
+[source,java]
+----
+public class Employee {
+    private long id;
+    private String name;
+}
+----
+--
+
+.NET::
++
+--
+[source,cs]
+----
+
+----
+--
+
+C++::
++
+--
+[source,cpp]
+----
+struct employee
+{
+    int64_t id;
+    std::string name;
+};
+----
+--
+
+Node.js::
++
+--
+[source,javascript]
+----
+class Employee {
+    id;
+    name;
+};
+----
+--
+
+Python::
++
+--
+[source,python]
+----
+class Employee:
+    def __init__(self, id: int, name: str):
+        self.id = id
+        self.name = name
+----
+--
+
+Go::
++
+--
+[source,go]
+----
+
+----
+--
+
+====
+
+
+[tabs]
+====
+
+Java::
++
+--
+[source,java]
+----
+public class Employee {
+    private long id;
+    private String name;
+    private int age; // Newly added field
+}
+----
+--
+
+.NET::
++
+--
+[source,cs]
+----
+
+----
+--
+
+C++::
++
+--
+[source,cpp]
+----
+struct employee
+{
+    int64_t id;
+    std::string name;
+    int32_t age;
```

Comment:
@OzanCansel can we please add the comments present in Java and Python code snippets here, to make them consistent? Note that, there are a few other code snippets down below, which lacks comments in C++